src/main/java/com/xiao/users/service/impl/UserServiceImpl.java

```diff
@@ -48,4 +45,21 @@ public Page<UserDto> findAllUser(int pages, int pageSize) {
         return usersPage.map(userMapper::userToUserDto);
     }
 
+    @Override
+    public UserDto updateUser(Long userId, UserDto userDto){
+        userDto.setId(userId);
+        userRepository.findById(userId).orElseThrow(
+                () -> new ResourceNotFoundException("User", "id", String.valueOf(userId))
+        );
+
+        if(userDto.getUsername().isEmpty() && userDto.getEmailAddress().isEmpty() && userDto.getPassword().isEmpty()){
```

Comment:
Don't perform validation manually like this! 
Let's introduce a new class called UpdateUserDto and create a custom validation annotation for updating user validation purposes. 
Please refer to this article: https://www.baeldung.com/spring-mvc-custom-validator